openstack/02a-MAClow/IEEE802154E.c

```diff
@@ -1705,8 +1717,14 @@ port_INLINE void incrementAsnOffset() {
          ieee154e_vars.asn.byte4++;
       }
    }
+   
    // increment the offsets
-   ieee154e_vars.slotOffset  = (ieee154e_vars.slotOffset+1)%schedule_getFrameLength();
+   frameLength = schedule_getFrameLength();
+   if (frameLength != 0) {
+      ieee154e_vars.slotOffset  = (ieee154e_vars.slotOffset+1)%frameLength;
+   } else {
+      ieee154e_vars.slotOffset  = ieee154e_vars.slotOffset + 1;

```

Comment:
ieee154e_vars.slotoffset is used in many prints to openserial: at the beginning (when a mote is not synchronized) the slotoffset is simply incremented as the ASN; of couse, the value of the slotOffset will wrap around to 0 after about 16 minutes with a 15ms timeslot (for this reason, when a mote is set to be DAGroot by openvisualizer, the slotOffset is recomputed by ieee154e_syncSlotOffset on the basis of the ASN which wraps around slowlier), but we can see errors and maybe relate them in time.
++ has been done.